src/util/psbt/map/input.rs

```diff
@@ -184,8 +184,19 @@ impl PsbtSigHashType {
         }
     }
 
-    /// Obtains the inner sighash byte from this [`PsbtSigHashType`].
-    pub fn inner(self) -> u32 {
+    /// Creates a [`PsbtSigHashType`] from a raw `u32`.
+    ///
+    /// Allows construction of a non-standard or non-valid sighash flag
+    /// (`EcdsaSigHashType`, `SchnorrSigHashType` respectively).
```

Comment:
Nit: please not block because of this one

```suggestion
    /// ([`EcdsaSigHashType`], [`SchnorrSigHashType`] respectively).
```

Reply:
Good catch, fixed as suggested.